### DISCUSSION TITLE
Why enforcing EIGEN_RUNTIME_NO_MALLOC ?

### DISCUSSION THREAD

**Discussion by jeanchristopheruel** [kul-optec/alpaqa#8] [General]
I would appreciate some clarification regarding the purpose of enforcing in [src/CMakeLists.txt](src/CMakeLists.txt), see [this line](https://github.com/kul-optec/alpaqa/blob/579f380f71fc708611ba8991da392e280a2f6a28/src/CMakeLists.txt#L58). My suggestion is to consider leaving this enforcement decision to the developer's discretion, as it doesn't appear to be a mandatory requirement for Alpaqa's implementation.

Thanks a lot!

**Comment by tttapa** [comment]
68c79768be1b7ce73f67df49a9bfde580e956dfd now disables these Eigen options by default. The checks can be enabled again by setting `ALPAQA_DEBUG_CHECKS_EIGEN=On`.

Note that `EIGEN_RUNTIME_NO_MALLOC` does not disable malloc. It simply adds a function that can be called to _temporarily_ disable allocations for a specific piece of code. It is used in debug mode only, and verifies that no unnecessary allocations take place in the inner loops of the algorithms.  
`EIGEN_INITIALIZE_MATRICES_BY_NAN` is very useful to check for uninitialized matrices/vectors.

Granted, these flags could cause ABI issues and ODR violations when combining alpaqa with other code that uses Eigen without the same flags, so disabling them by default and leaving it up to the user is indeed a good idea, thanks!

**Reply by jeanchristopheruel** [reply]
Thanks a lot for the clarification! Indeed, it makes a lot of sense for debug purposes.
Closing discussion.
